lwip/example-lwip-native-echoclient.c

```diff
@@ -0,0 +1,224 @@
+#include <wolfcrypt/test/test.h>
+
+static err_t tcp_echoclient_connected(void *arg, struct tcp_pcb *tpcb, err_t err);
+static void TLS_shutdown(void);
+static int TLS_connect(void);
+static int TLS_setup(void);
+
+struct tcp_pcb *tls_pcb   = NULL;
+struct tcp_pcb *debug_pcb = NULL;
+
+static int tlsConnected = 0;
+static int tcpConnected = 0;
+static int tlsWaitingForReply = 0;
+
+static WOLFSSL_CTX *ctx = NULL;
+static WOLFSSL *ssl = NULL;
+
+void loggingCb(const int logLevel, const char *const logMessage)
+{
+    printf("%s\n", logMessage);
+    (void)logLevel;
+}
+
+void tls_echoclient_connect(void)
+{
+    ip_addr_t DestIPaddr;
+
+    if (tcpConnected == 0) {
+        tls_pcb = tcp_new();
+        if (tls_pcb != NULL) {
+            IP4_ADDR( &DestIPaddr, DEST_IP_ADDR0, DEST_IP_ADDR1, DEST_IP_ADDR2, DEST_IP_ADDR3 );
+            tcp_connect(tls_pcb,&DestIPaddr,DEST_PORT,tcp_echoclient_connected);
+        }
+    }
+
+    if (tcpConnected == 1 && tlsConnected == 0) {
+        TLS_connect();
+    }
+
+    if (tlsConnected == 1) {
+        int ret;
+        char reply[256];
+        int err;
+
+        if (tlsWaitingForReply == 0) {
+            ret = wolfSSL_write(ssl, "TLS **TEST 1** ", sizeof("TLS **TEST 1** "));
+            if (ret <= 0) {
+                loggingCb(0, "error when writing TLS message");
+                shutdown();
+            }
+            else {
+                tlsWaitingForReply = 1;
+            }
+        }
+
+        /* read a reply from the server */
+        if (tlsWaitingForReply == 1) {
+            memset(reply, 0, sizeof(reply));
+            ret = wolfSSL_read(ssl, reply, sizeof(reply));
+            if (ret <= 0) {
+                err = wolfSSL_get_error(ssl, 0);
+                if (err != SSL_ERROR_WANT_READ &&
+                    err != WOLFSSL_ERROR_WANT_WRITE) {
+                    loggingCb(0, "error when reading TLS message");
+                    shutdown();
+                }
+            }
+            if (ret > 0) {
+                loggingCb(0, reply);
+                shutdown(); /* received reply, done with connection */
+            }
+        }
+    }
+}
+
+static err_t tcp_echoclient_connected(void *arg, struct tcp_pcb *tpcb, err_t err)
+{
+    if (err == ERR_OK) {
+        tcpConnected = 1;
+        /* a tcp_poll could be added here for connection timeout checking */
+        return TLS_setup();
+    }
+    return ERR_OK;
+}
+
+
+/* setup all the TLS structures and load CA */
+static int TLS_setup(void)
+{
+    int ret;
+
+#ifdef DEBUG_WOLFSSL
+    /* redirect wolfssl debug print outs */
+    wolfSSL_SetLoggingCb(loggingCb);
+    wolfSSL_Debugging_ON();
+#endif
+    wolfSSL_Init(); /* called once on program startup */
+    ctx = wolfSSL_CTX_new(wolfTLSv1_2_client_method());
+    if (ctx == NULL) {
+        loggingCb(0, "ctx was null");
+        shutdown();
+        return ERR_MEM;
+    }
+
+    ret = wolfSSL_CTX_load_verify_buffer(ctx, CA_CERTS, CA_CERTS_LEN,

```

Comment:
Since you're setting up the demo to work with (Linux? Mac? Win?) edition of wolfssl and user is getting this wolfSSL via STM32CubeIDE there is a good chance of certificate chain mismatching.  I would suggest not verifying the chain thus mitigating a potential pain point for the user.